src/dmd/target.d

```diff
@@ -774,6 +775,38 @@ struct Target
         const sz = t.size(loc);
         return global.params.is64bit ? (sz + 7) & ~7 : (sz + 3) & ~3;
     }
+
+    /**
+     * Get targetInfo by key
+     * Params:
+     *  name = name of targetInfo to get
+     *  loc = location to use for error messages
+     * Returns:
+     *  Expression for the requested targetInfo
+     */
+    extern (C++) static Expression getTargetInfo(const(char)* name, const ref Loc loc)
+    {
+        switch (name.toDString)
+        {
+            case "cxxlib":
+                return new StringExp(loc, global.params.mscrtlib);
+            default:
+                break;
+        }
+        return new ErrorExp();
+    }
+
+    /**
+     * Get all supported target-info keys
+     * Returns:
+     *  Array of targetInfo keys
+     */
+    extern (C++) static const(char)** allTargetInfos(size_t* numTargetInfos)
```

Comment:
Are we allowed to use `out` parameters?

Reply:
Would be nice if D strings and arrays can be used.

Reply:
These have to go in the .h file... i figured i need to stick with in C++ for these functions?

Reply:
Of course, I'd love to use dstrings and arrays!

Reply:
Now that I think about the ABI is not compatible on Windows for returning D arrays.